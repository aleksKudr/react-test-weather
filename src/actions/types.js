export const FETCH_WEATHER = 'FETCH_WEATHER';
export const FETCH_HISTORY = 'FETCH_HISTORY';
export const CREATE_WEATHER = 'CREATE_WEATHER';
export const SIGN_IN = 'SIGN_IN';
export const SIGN_OUT = 'SIGN_OUT';
export const CREATE_USER = 'CREATE_USER';