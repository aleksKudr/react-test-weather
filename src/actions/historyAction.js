// import axios from 'axios';
import { instance } from '../utils/axios';
import {
    FETCH_HISTORY
} from './types';
 
export const getHistory = () => async dispatch => {
    const response = await instance.get(`/history`);
    // console.log('response2', response);
    dispatch({ type: FETCH_HISTORY, payload: response});
    // return true;
}