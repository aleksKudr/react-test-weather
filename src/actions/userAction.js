// import streams from '../apis/streams';
// import history from '../history';
import { instance } from '../utils/axios';
import {
  SIGN_IN,
  SIGN_OUT,
  CREATE_USER
} from './types';
import { setToken } from '../utils/functions';

export const regist = formValues => async (dispatch, getState) => {
  const response = await instance.post('/regist', {
    username: formValues.username,
    password: formValues.password
  });
  setToken(response.data.token);

  dispatch({ type: CREATE_USER,
    payload: formValues });
};

export const login = formValues => async (dispatch) => {
  console.log('formValues', JSON.stringify(formValues));
  const response = await instance.post('login', {
    username: formValues.username,
    password: formValues.password
  });


  setToken(response.data.token);
  console.log(formValues);
  //   console.log('response', JSON.stringify(response));
  dispatch({ type: SIGN_IN,
    payload: formValues });
};

export const logout = () => async (dispatch) => {
  dispatch({ type: SIGN_OUT });
};
