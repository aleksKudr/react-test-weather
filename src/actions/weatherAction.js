// import streams from '../apis/streams';
// import history from '../history';
import axios from 'axios';
import { instance } from '../utils/axios';
import {
  FETCH_WEATHER,
  CREATE_WEATHER
} from './types';

export const createHistory = data => async (dispatch) => {
  console.log('data', data);
  const response = await instance.post('/history', {
    data
  });
  console.log('response1', response);
  dispatch({ type: CREATE_WEATHER,
    payload: response });
};

export const getWeather = formValues => async (dispatch) => {
  const KEY = '36e48c55215089bf93e20c56ac459509';
  // console.log(formValues)
  const response = await axios.get('http://api.openweathermap.org/data/2.5/forecast', {
    params: {
      apiKey: KEY,
      q: formValues.city
    }
  });
  console.log('response', JSON.stringify(response));

  dispatch({ type: FETCH_WEATHER,
    payload: response });
};
