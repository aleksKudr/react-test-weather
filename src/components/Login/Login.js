import React, {
  PureComponent } from 'react';
import './Login.css';
import { connect } from 'react-redux';
import { getToken } from '../../utils/functions';
import Header from '../Header';
import { login } from '../../actions/userAction';
import LoginForm from './LoginForm';


export class Login extends PureComponent {
  componentWillMount() {
    if (getToken())
    { this.props.history.replace('/'); }
  }

  onSubmit = async (formValues) => {
    console.log('this.props', this.props);
    await this.props.login(formValues);

    if (getToken()) {
      this.props.history.replace('/');
    }

    // this.props.history.replace('/');

  }




  render() {
    return (
      <div>
        <Header />
        <div className="center">
          <div className="card">
            <div>
              <h3>Login</h3>
              <LoginForm onSubmit={this.onSubmit} />
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default connect(
  null,
  { login }
)(Login);


// import Header from '../Header';
// import {login} from '../../utils/axios';

// class Login extends Component {
//     constructor() {
//         super();
//         this.handleChange = this.handleChange.bind(this);
//         this.handleFormSubmit = this.handleFormSubmit.bind(this);
//         // this.Auth = new AuthService();
//     }
//     componentWillMount() {
//         if (getToken())
//             this.props.history.replace('/');
//     }
//     render() {
//         return (<div><Header/>
//             <div className="center">
//                 <div className="card">

//                     <h1>Login</h1>
//                     <form onSubmit={this.handleFormSubmit}>
//                         <input
//                             className="form-item"
//                             placeholder="Username goes here..."
//                             name="username"
//                             type="text"
//                             onChange={this.handleChange}
//                             />
//                         <input
//                             className="form-item"
//                             placeholder="Password goes here..."
//                             name="password"
//                             type="password"
//                             onChange={this.handleChange}
//                             />
//                         <input
//                             className="form-submit"
//                             value="SUBMIT"
//                             type="submit"
//                             />

//                     </form>
//                 </div>
//             </div>
//             </div>
//         );
//     }

//     handleFormSubmit(e) {
//         e.preventDefault();

//         login(this.state.username, this.state.password)
//             .then(res => {
//                 this.props.history.replace('/');
//             })
//             .catch(err => {
//                 alert(err);
//             })
//     }

//     handleChange(e) {
//         this.setState({
//             [e.target.name]: e.target.value
//         })
//     }
// }

// export default Login;
