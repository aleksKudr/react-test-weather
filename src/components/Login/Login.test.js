import React from 'react';
import configureStore from 'redux-mock-store';
import Header from '../Header';
import LoginForm from './LoginForm';
import { Login } from './Login';
import { login } from '../../actions/userAction';


it('test Header', () => {
  const wrapper = shallow(
    <Header />
  );

  //   let tree = component.toJSON();
  expect(wrapper).toMatchSnapshot();
});
it('test h3', () => {
  const wrapper = shallow(
    <h3 />
  );

  //   let tree = component.toJSON();
  expect(wrapper).toMatchSnapshot();
});
it('test RegistForm', () => {
  const wrapper = shallow(
    <LoginForm onSubmit={jest.fn()} />
  );

  //   let tree = component.toJSON();
  expect(wrapper).toMatchSnapshot();
});
// it('test RegistForm', () => {
//   const wrapper = shallow(
//     <Login login={login()} />
//   );
//   const spyOn = jest.spyOn(wrapper.instance(), 'onSubmit');
//   wrapper.instance().forceUpdate();
//   wrapper
//     .find(LoginForm)
//     .at(0)
//     .simulate('click');
//   expect(spyOn).toHaveBeenCalled();

//   //   let tree = component.toJSON();
//   // expect(wrapper).toMatchSnapshot();
// });
const setup = () => {
  const wrapper = shallow(<Login
    history={{ replace: jest.fn() }}
    login={jest.fn()}
  />);
  return wrapper;
};
const param = {
  username: 'adm',
  password: 'adm'
};

test('has access to `success` state', () => {
  const wrapper = setup();
  // console.log('wrapper.debug()', wrapper.debug());
  const spyOn = jest.spyOn(wrapper.instance(), 'onSubmit');
  wrapper.instance().forceUpdate();
  const test = wrapper
    .find(LoginForm)
    // .childAt(0)
    .dive()
    .simulate('submit', param);

  // console.log('test.debug()', test.debug());
  // console.log('wrapper..instance.debug()', wrapper.debug());

  expect(spyOn).toHaveBeenCalledWith(param);
  expect(wrapper.instance().props.login).toHaveBeenCalled();
  expect(wrapper.instance().props.login).toHaveBeenCalledWith(param);
});
