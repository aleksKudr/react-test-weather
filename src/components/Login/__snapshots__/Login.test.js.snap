// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Header 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Navbar
        color="light"
        expand="md"
        light={true}
        tag="nav"
      >
        <Nav
          className="ml-auto"
          navbar={true}
          tag="ul"
          vertical={false}
        >
          <NavItem
            tag="li"
          >
            <NavLink
              href="/login"
              tag="a"
            >
              Login
            </NavLink>
          </NavItem>
          <NavItem
            tag="li"
          >
            <NavLink
              href="/regist"
              tag="a"
            >
               Regist
            </NavLink>
          </NavItem>
        </Nav>
      </Navbar>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Nav
          className="ml-auto"
          navbar={true}
          tag="ul"
          vertical={false}
        >
          <NavItem
            tag="li"
          >
            <NavLink
              href="/login"
              tag="a"
            >
              Login
            </NavLink>
          </NavItem>
          <NavItem
            tag="li"
          >
            <NavLink
              href="/regist"
              tag="a"
            >
               Regist
            </NavLink>
          </NavItem>
        </Nav>,
        "color": "light",
        "expand": "md",
        "light": true,
        "tag": "nav",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <NavItem
              tag="li"
            >
              <NavLink
                href="/login"
                tag="a"
              >
                Login
              </NavLink>
            </NavItem>,
            <NavItem
              tag="li"
            >
              <NavLink
                href="/regist"
                tag="a"
              >
                 Regist
              </NavLink>
            </NavItem>,
          ],
          "className": "ml-auto",
          "navbar": true,
          "tag": "ul",
          "vertical": false,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <NavLink
                href="/login"
                tag="a"
              >
                Login
              </NavLink>,
              "tag": "li",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Login",
                "href": "/login",
                "tag": "a",
              },
              "ref": null,
              "rendered": "Login",
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <NavLink
                href="/regist"
                tag="a"
              >
                 Regist
              </NavLink>,
              "tag": "li",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": " Regist",
                "href": "/regist",
                "tag": "a",
              },
              "ref": null,
              "rendered": " Regist",
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Navbar
          color="light"
          expand="md"
          light={true}
          tag="nav"
        >
          <Nav
            className="ml-auto"
            navbar={true}
            tag="ul"
            vertical={false}
          >
            <NavItem
              tag="li"
            >
              <NavLink
                href="/login"
                tag="a"
              >
                Login
              </NavLink>
            </NavItem>
            <NavItem
              tag="li"
            >
              <NavLink
                href="/regist"
                tag="a"
              >
                 Regist
              </NavLink>
            </NavItem>
          </Nav>
        </Navbar>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Nav
            className="ml-auto"
            navbar={true}
            tag="ul"
            vertical={false}
          >
            <NavItem
              tag="li"
            >
              <NavLink
                href="/login"
                tag="a"
              >
                Login
              </NavLink>
            </NavItem>
            <NavItem
              tag="li"
            >
              <NavLink
                href="/regist"
                tag="a"
              >
                 Regist
              </NavLink>
            </NavItem>
          </Nav>,
          "color": "light",
          "expand": "md",
          "light": true,
          "tag": "nav",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <NavItem
                tag="li"
              >
                <NavLink
                  href="/login"
                  tag="a"
                >
                  Login
                </NavLink>
              </NavItem>,
              <NavItem
                tag="li"
              >
                <NavLink
                  href="/regist"
                  tag="a"
                >
                   Regist
                </NavLink>
              </NavItem>,
            ],
            "className": "ml-auto",
            "navbar": true,
            "tag": "ul",
            "vertical": false,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <NavLink
                  href="/login"
                  tag="a"
                >
                  Login
                </NavLink>,
                "tag": "li",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Login",
                  "href": "/login",
                  "tag": "a",
                },
                "ref": null,
                "rendered": "Login",
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <NavLink
                  href="/regist"
                  tag="a"
                >
                   Regist
                </NavLink>,
                "tag": "li",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": " Regist",
                  "href": "/regist",
                  "tag": "a",
                },
                "ref": null,
                "rendered": " Regist",
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`test RegistForm 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ForwardRef
    destroyOnUnmount={true}
    enableReinitialize={false}
    forceUnregisterOnUnmount={false}
    form="streamForm"
    getFormState={[Function]}
    keepDirtyOnReinitialize={false}
    onSubmit={[MockFunction]}
    persistentSubmitErrors={false}
    pure={true}
    shouldAsyncValidate={[Function]}
    shouldError={[Function]}
    shouldValidate={[Function]}
    shouldWarn={[Function]}
    touchOnBlur={true}
    touchOnChange={false}
    updateUnregisteredFields={false}
    validate={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "destroyOnUnmount": true,
      "enableReinitialize": false,
      "forceUnregisterOnUnmount": false,
      "form": "streamForm",
      "forwardedRef": null,
      "getFormState": [Function],
      "keepDirtyOnReinitialize": false,
      "onSubmit": [MockFunction],
      "persistentSubmitErrors": false,
      "pure": true,
      "shouldAsyncValidate": [Function],
      "shouldError": [Function],
      "shouldValidate": [Function],
      "shouldWarn": [Function],
      "touchOnBlur": true,
      "touchOnChange": false,
      "updateUnregisteredFields": false,
      "validate": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "destroyOnUnmount": true,
        "enableReinitialize": false,
        "forceUnregisterOnUnmount": false,
        "form": "streamForm",
        "forwardedRef": null,
        "getFormState": [Function],
        "keepDirtyOnReinitialize": false,
        "onSubmit": [MockFunction],
        "persistentSubmitErrors": false,
        "pure": true,
        "shouldAsyncValidate": [Function],
        "shouldError": [Function],
        "shouldValidate": [Function],
        "shouldWarn": [Function],
        "touchOnBlur": true,
        "touchOnChange": false,
        "updateUnregisteredFields": false,
        "validate": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`test h3 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <h3 />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": "h3",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": "h3",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
